src/ripple/net/impl/HTTPDownloader.cpp

```diff
@@ -140,22 +126,25 @@ HTTPDownloader::do_session(
     // because the server is shutting down,
     // this method notifies a 'Stoppable'
     // object that the session has ended.
-    auto exit = [this]() {
+    auto exit = [this, &dstPath, complete](bool invokeComplete) {
+        if (invokeComplete)
```

Comment:
The only time ```complete``` isn't called is when  ```cancelDownloads_``` is true.
Consider this simplification...
```
    auto exit = [this, &dstPath, complete]() {
        if (!cancelDownloads_)
            complete(std::move(dstPath));

        std::lock_guard<std::mutex> lock(m_);
        sessionActive_ = false;
        c_.notify_one();
    };
```

Reply:
Consider renaming ```cancelDownloads_``` to ```stop_```

Reply:
Done